akka-serialization-jackson/src/main/scala/akka/serialization/jackson/JacksonSerializer.scala

```diff
@@ -444,44 +489,56 @@ import akka.util.Helpers.toRootLowerCase
     (fromVersion, manifestClassName)
   }
 
+  private lazy val lz4Factory = LZ4Factory.fastestInstance()
+  private lazy val lz4Compressor = lz4Factory.fastCompressor()
+  private lazy val lz4Decompressor = lz4Factory.safeDecompressor()
```

Comment:
Could perhaps be better to create these up front based on compression algorithm instead of lazily since lazy uses synchronization? Also, I don't think we have to hang on to the lz4Factory once we created the compressor and decompressor, right?

Reply:
I am not sure how to accomplish this. We need a value which is lazily evaluated and memorized without synchronization. I don't know any way to do that without synchronization. Could you please give an example?

Reply:
They could be needed for deserialization also if compression isn't enabled. I think we can just keep it like this. We have some other `lazy` in this class anyway.

Would be good to move them more to the top of the class though, before `def manifest`